src/charm.py

```diff
@@ -548,6 +548,17 @@ def _on_update_status(self, event: UpdateStatusEvent):
                 self.unit.status = WaitingStatus("Waiting for MongoDB to start")
                 return
 
+        # Cannot check more advanced MongoDB statuses if the cluster doesn't have passwords synced
+        # this can occur in two cases:
+        # 1. password rotation
+        # 2. race conditions when a new shard is addeded.
+        if (
+            not self.shard.cluster_password_synced()
+            or not self.config_server.cluster_password_synced()
```

Comment:
Is update_status the only hook where this must be handled?

Reply:
It is the only place I have seen the error occur, but I am surprised I have not seen it elsewhere.

I could be easily convinced to add it elsewhere in the code, but I believe this is sufficent for the time-being